config/crd/snapshot.storage.k8s.io_volumesnapshotcontents.yaml

```diff
@@ -11,155 +11,160 @@ spec:
+                If not specified, the default will be VolumeSnapshotContentRetain
+                for static binding, and VolumeSnapshotContentDelete for dynamic snapshot
+                creation.
+              type: string
+            persistentVolumeRef:
+              description: PersistentVolumeRef represents the PersistentVolume that
+                the snapshot has been taken from. In dynamic snapshot creation case,
+                the field will be specified when VolumeSnapshot and VolumeSnapshotContent
+                are bound.
+              properties:
+                apiVersion:
+                  description: API version of the referent.
+                  type: string
+                fieldPath:
+                  description: 'If referring to a piece of an object instead of an
+                    entire object, this string should contain a valid JSON/Go field
+                    access statement, such as desiredState.manifest.containers[2].
+                    For example, if the object reference is to a container within
+                    a pod, this would take on a value like: "spec.containers{name}"
+                    (where "name" refers to the name of the container that triggered
+                    the event) or if no container name is specified "spec.containers[2]"
+                    (container with index 2 in this pod). This syntax is chosen only
+                    to have some well-defined way of referencing a part of an object.
+                    TODO: this design is not final and this field is subject to change
+                    in the future.'
+                  type: string
+                kind:
+                  description: 'Kind of the referent. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+                  type: string
+                name:
+                  description: 'Name of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names'
+                  type: string
+                namespace:
+                  description: 'Namespace of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/'
+                  type: string
+                resourceVersion:
+                  description: 'Specific resourceVersion to which this reference is
+                    made, if any. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#concurrency-control-and-consistency'
+                  type: string
+                uid:
+                  description: 'UID of the referent. More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids'
+                  type: string
+              type: object
+            volumeSnapshotRef:
+              description: VolumeSnapshotRef is part of bi-directional binding between
```

Comment:
"volumeSnapshotRef specifies the VolumeSnapshot object that this VolumeSnapshotContent is associated with. The VolumeSnapshot.Spec.VolumeSnapshotContentName field must also reference this VolumeSnapshotContent name for the association to be considered valid.  If the referenced VolumeSnapshot object does not exist, then the VolumeSnapshotContent.Spec.DeletionPolicy is triggered."

I think it's also important to talk about UID and when and when not to set it. But I haven't thought of a good way to explain it yet.

Reply:
that sounds exposing a lot of controller implementation details in for this field. I will rephrase the uid piece.